Lib/idlelib/help.py

```diff
@@ -150,6 +157,69 @@ def handle_data(self, data):
             self.text.insert('end', d, (self.tags, self.chartags))
 
 
+class FontSizer:
+    "Support dynamic widget font resizing."
+    def __init__(self, widget, callback=None):
+        """"Add font resizing functionality to widget.
+
+        Args:
+            widget: Tk widget with font attribute to size.
+            callback: Function to call for additional font resizing
```

Comment:
It was designed to work without a callback for a just a text widget with a font and I had tried to write the `FontSizer` tests in a way that showed that (that is, the test didn't define a callback).  

The callback was written in order to allow for scaling of the other size fonts defined in `help.py`.  The `test_scale_tagfonts` and `test_resizing_callback` were meant to show that.  Sorry it wasn't clear.

Here was the issue:
In `help.py`, there are several different tags (h3, h2, h1) that have, by definition, different sizes.  So, I needed a way to change all the font sizes when the event was triggered.  With FontSizer being extracted as its own class, that class would only operate on the font defined within the text widget and not other ancillary fonts somehow related to that widget (in tags).  
Options:
1. What I really wanted to do was to have the reassignment of the font size in FontSizer be the trigger event.  That way when the font size changed, the HelpText class could watch for that event and call `scale_tagfonts`.  Even though there is a `Configure` event available, it seemed more to be for window resizing rather than font changes.
2.  A second option I considered was to add an event timer like in CodeContext, but I thought this wouldn't be smooth since one font would change immediately, but the others would be on a (possible) delay.  Plus, you didn't really like the timers in CodeContext.
3.  Add a callback that can be sent to FontSizer for any widgets that needed to do their own processing after the main font was changed.  This was what I implemented.
4.  Thinking about it now, I've come up with another idea.  Maybe there's a way for FontSizer to look at all the tags defined to `text` and modify the fonts for those tags.  I'll have to see if that's possible.

Did you have other ideas in mind?